File: src/shared/contracts/import.ts
import type { LanguageCode, PackId, PreviewResult, PreviewToken, ValidationIssue, WorkspaceId } from "./common";
import type { JobAccepted } from "./job";

export interface PreviewImportPackInput {
  workspaceId: WorkspaceId;
  newPackName: string;
  newPackCode: string | null;
  newPackAuthor: string;
  newPackVersion: string;
  newPackDescription: string | null;
  displayLanguageOrder: LanguageCode[];
  defaultExportLanguage: LanguageCode | null;
  cdbPath: string;
  picsDir: string | null;
  fieldPicsDir: string | null;
  scriptDir: string | null;
  stringsConfPath: string | null;
  sourceLanguage: LanguageCode;
}

export interface ImportPreview {
  target_pack_id: PackId;
  target_pack_name: string;
  card_count: number;
  warning_count: number;
  error_count: number;
  missing_main_image_count: number;
  missing_script_count: number;
  missing_field_image_count: number;
  issues: ValidationIssue[];
}

export interface ExecuteImportPackInput {
  previewToken: PreviewToken;
}

export type ImportPreviewResult = PreviewResult<ImportPreview>;
export type ExecuteImportPackResult = JobAccepted;

File: src-tauri/src/application/dto/import.rs
use std::path::PathBuf;

use serde::{Deserialize, Serialize};

use crate::domain::common::ids::{LanguageCode, PackId, PreviewToken, WorkspaceId};
use crate::domain::common::issue::ValidationIssue;

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct PreviewImportPackInput {
    pub workspace_id: WorkspaceId,
    pub new_pack_name: String,
    pub new_pack_author: String,
    pub new_pack_version: String,
    pub new_pack_description: Option<String>,
    pub display_language_order: Vec<LanguageCode>,
    pub default_export_language: Option<LanguageCode>,
    pub cdb_path: PathBuf,
    pub pics_dir: Option<PathBuf>,
    pub field_pics_dir: Option<PathBuf>,
    pub script_dir: Option<PathBuf>,
    pub strings_conf_path: Option<PathBuf>,
    pub source_language: LanguageCode,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
pub struct ImportPreviewDto {
    pub target_pack_id: PackId,
    pub target_pack_name: String,
    pub card_count: usize,
    pub warning_count: usize,
    pub error_count: usize,
    pub missing_main_image_count: usize,
    pub missing_script_count: usize,
    pub missing_field_image_count: usize,
    pub issues: Vec<ValidationIssue>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ExecuteImportPackInput {
    pub preview_token: PreviewToken,
}
